mt_sat/mt_sat_wrapper.m

```diff
@@ -221,6 +237,24 @@ function mt_sat_wrapper(mtw_nii,pdw_nii,t1w_nii,mtw_jsn,pdw_jsn,t1w_jsn,varargin
     savejson('',provenance,[pwd filesep 'MTsat.json']);
 end
 
+% JSON file for dataset_description
+addDescription = struct();
+addDescription.Name = 'qMRLab Outputs';
+addDescription.BIDSVersion = '1.5.0';
+addDescription.DatasetType = 'derivative';
+addDescription.GeneratedBy.Name = 'qMRLab';
+addDescription.GeneratedBy.Version = qMRLabVer;
+addDescription.GeneratedBy.Container.Type = 'docker (if Docker{enabled = true})';
+addDescription.GeneratedBy.Container.Tag = 'qmrlab/minimal:2.4.1';
+addDescription.GeneratedBy.Name2 = 'Manual';
+addDescription.GeneratedBy.Description = 'Generated example T1map outputs';
+addDescription.SourceDatasets.DOI = 'DOI 10.17605/OSF.IO/K4BS5';
```

Comment:
These fields are optional in the `dataset_decription.json`. We can manage these more elegantly as these won't be fixed.

Reply:
@agahkarakuzu sorry, I don't understand when you say those fields won't be fixed. When you say fixed, do you mean that a field will not change, (e.g. `addDescription.GeneratedBy.Description = 'Generated example T1map outputs'`)

Also, could you please point out which fields are going to be fixed, and which are not?

Reply:
You can find the partial answer to this question above. [Here](https://bids-specification.readthedocs.io/en/stable/03-modality-agnostic-files.html#dataset_descriptionjson) you can find out which ones are required, optional or recommended. 

I think you can infer which ones are subjected to change.

Reply:
@agahkarakuzu for `addDescription.SourceDatasets` I assigned variables with default values. The user can introduce the DOI, URL, Version of the dataset from the exec command. Does this work for you?

Reply:
@jvelazquez-reyes indeed, they can either set that metadata in the config or pass during execution. For the dataset fields, we can assign defaults empty and make wrapper ignore writing out keys for empty values.

Reply:
@agahkarakuzu I set dataset fields as defaults empty. Then, I made the wrapper to ignore those fields with something like this: `if ~isempty(p.Results.datasetDOI); addDescription.SourceDatasets.DOI = p.Results.datasetDOI; end`

It worked but let me know if you wanted another approach to define this behaviour.